javascript/i18n/phonenumbers/demo.js

```diff
@@ -63,57 +64,68 @@ function phoneNumberParser() {
         case PNV.TOO_LONG:
           output.append('TOO_LONG');
           break;
+        case PNV.INVALID_LENGTH:
+          output.append('INVALID_LENGTH');
+          break;
       }
       // IS_POSSIBLE shouldn't happen, since we only call this if _not_
       // possible.
       output.append('\nNote: numbers that are not possible have type ' +
           'UNKNOWN, an unknown region, and are considered invalid.');
     } else {
-      var isNumberValid = phoneUtil.isValidNumber(number);
-      output.append('\nResult from isValidNumber(): ');
-      output.append(isNumberValid);
-      if (isNumberValid && regionCode && regionCode != 'ZZ') {
+      // Checking as isValid() fails if possible local only.
+      if (isPossibleReason == PNV.IS_POSSIBLE_LOCAL_ONLY) {
+        output.append('\nResult from isPossibleNumberWithReason(): ');
+        output.append('IS_POSSIBLE_LOCAL_ONLY');
+        output.append('\nNumber is considered as invalid as it is ' +
+          'not a possible national number.');
+      } else {
+        var isNumberValid = phoneUtil.isValidNumber(number);
+        output.append('\nResult from isValidNumber(): ');
+        output.append(isNumberValid);
+        if (isNumberValid && regionCode && regionCode != 'ZZ') {
         output.append('\nResult from isValidNumberForRegion(): ');
```

Comment:
indentation looks off here

Reply:
Done. Made use of internal formatting tools. :)
Its painful without them

Reply:
You can set up vim to understand formatting rules, and then apply them to a file....